file/writable_file_writer.h

```diff
@@ -138,11 +139,11 @@ class WritableFileWriter {
  public:
   WritableFileWriter(
       std::unique_ptr<FSWritableFile>&& file, const std::string& _file_name,
-      const FileOptions& options, Env* env = nullptr,
-      Statistics* stats = nullptr,
+      const FileOptions& options, std::shared_ptr<IOTracer> io_tracer,
```

Comment:
Rather than passing the IOTracer here, would it be better to add the IOTrace to the Env/FileOptions?  This would eliminate the need for the Ptr classes, as you could simply return the proper "Writer"